pkg/machines/components/storagePools/createStoragePoolDialog.jsx

```diff
@@ -0,0 +1,327 @@
+    );
+};
+
+const StoragePoolConnectionRow = ({ onValueChanged, dialogValues, loggedUser }) => {
+    let connectionUris = [
+        <Select.SelectEntry data={LIBVIRT_SYSTEM_CONNECTION}
+                            key={LIBVIRT_SYSTEM_CONNECTION}>{_("QEMU/KVM System connection")}
+        </Select.SelectEntry>,
+    ];
+
+    // Root user should not be presented the session connection
+    if (loggedUser.id != 0)
+        connectionUris.push(
+            <Select.SelectEntry data={LIBVIRT_SESSION_CONNECTION}
+                key={LIBVIRT_SESSION_CONNECTION}>{_("QEMU/KVM User connection")}
+            </Select.SelectEntry>
+        );
+
+    return (
+        <FormGroup controlId='connection'>
+            <Grid.Col componentClass={Form.ControlLabel} sm={3}>
+                {_("Connection")}
+            </Grid.Col>
+            <Grid.Col sm={6}>
+                <Select.Select id='storage-pool-input-type'
+                               initial={dialogValues.connectionName}
+                               onChange={value => onValueChanged('connectionName', value)}>
+                    {connectionUris}
+                </Select.Select>
+            </Grid.Col>
+        </FormGroup>
+    );
+};
+
+const StoragePoolSourceRow = ({ onValueChanged, dialogValues }) => {
+    if (['netfs'].includes(dialogValues.type))
+        return (
+            <FormGroup controlId='source'>
+                <Grid.Col componentClass={Form.ControlLabel} sm={3}>
+                    {_("Source Path")}
+                </Grid.Col>
+                <Grid.Col sm={6}>
+                    <input id='storage-pool-input-source'
+                           type='text'
+                           minLength={1}
+                           placeholder={_("The directory on the server being exported")}
+                           value={dialogValues.source.dir || ''}
+                           onChange={e => onValueChanged('source', {'dir': e.target.value})}
+                           className='form-control' />
+                </Grid.Col>
+            </FormGroup>
+        );
+    return null;
+};
+
+const StoragePoolTargetRow = ({ onValueChanged, dialogValues }) => {
+    if (['dir', 'netfs'].includes(dialogValues.type))
+        return (
+            <FormGroup controlId='target'>
+                <Grid.Col componentClass={Form.ControlLabel} sm={3}>
+                    {_("Target Path")}
+                </Grid.Col>
+                <Grid.Col sm={6}>
+                    <FileAutoComplete.FileAutoComplete
+                        placeholder={_("Path on host's filesystem")}
+                        onChange={value => onValueChanged('target', value)} />
+                </Grid.Col>
+            </FormGroup>
+        );
+};
+
+const StoragePoolAutostartRow = ({ onValueChanged, dialogValues }) => {
+    return (
+        <FormGroup controlId='autostart'>
+            <Grid.Col componentClass={Form.ControlLabel} sm={3}>
+                {_("Autostart")}
```

Comment:
What does Autostart mean in this case? Does it start when the libvirtd service start, or something else?

Reply:
It determines whether the pool is automatically started at boot time.

Reply:
How about something like this for a bit more clarity:
"**Startup** [x] Start pool when host boots"

Reply:
I like the help message, but if you don't mind I 'd rather keep the label name to autostart since it's used for other resources as well, and it matches the libvirt terminology with which some people might already be familiar with.

Reply:
What about "**Startup** [x] Autostart pool on boot"
That way it explains a little better, but still keeps the familiar term **Autostart**.
Since we're primarily targeting new linux admins (including senior Windows admins), we need something that works for people who are not previously familiar with libvirt.

It also places the checkbox on the left of the label it's connected to, and that's the way checkboxes works everywhere else. Not super-happy about the label **Startup**, but it was the best I could come up with.